[Link] Add a Link component (#14093)
mui-org/material-ui: docs/src/pages/style/links/links.md

---
components: Link
---

# Links

> <p class="description">The Link component allows you to show links on the page, open them in a new window.</p>

Reviewer: ```suggestion
<p class="description">The Link component allows you to show a link on the page, and opens it in a new window.</p>
```

Author: By default, links don't open in a new window

Reviewer 2: Yeah, my wording is crap.

Reviewer: Then it probably shouldn't say that it does! (I was just correcting the grammar)...

Having said that, would a prop to do so optionally be a good idea? Looking at the benchmarks @oliviertassinari provided, this one is closest:

pinterest.github.io/gestalt/#/Link

It might also be helpful to explain when to use Link rather than `<a>`.

## Simple links

{{"demo": "pages/style/links/Links.js"}}

The Link component is built on top of the [Typography](/api/typography/) component.

> You can leverage its properties.

Reviewer: This is slightly confusing, as the demo that follows doesn't actually uses the properties of Typography, but rather wraps it in the Typography component, while the demo thet precees it does.

---
components: Link
---

# Links

<p class="description">The Link component creates a hyperlink to other web pages, files, locations within the same page, email addresses, or any other URL.</p>

> 

Reviewer: What I meant was, it might be helpful to explain why a developer should use the Link component rather than simply using an HTML anchor. I.e. what is unique the purpose of this component?

Author: I'll change it to that, I just thought this was a better description than what was there

## Simple links

The Link component is built on top of the [Typography](/api/typography/) component.
You can leverage its properties.

{{"demo": "pages/style/links/Links.js"}}

For the best user experience links should stand out from the text on the page.

> {{"demo": "pages/style/links/LinkUserExperience.js"}}

Reviewer: What's the purpose of this demo? The (non-inherit) links in the previous example already demonstrate this feature. It seems a bit redundant.

Author: It was more for accessibility, I've moved the sentence into the Accessibility section and removed the demo

## Accessibility

- When providing the content for the link, avoid phrases like "click here" or "go to".
- If a link doesn't have a meaningful href, [it should be rendered using a `<button>` element](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md).

{{"demo": "pages/style/links/ButtonLink.js"}}

## Recommendations

> When you use `target="_blank"` with Links it is recommended to always use `rel="noopener"` or `rel="noreferrer"`

Reviewer: Only if linking to third-party content...

`rel="noopener"` prevents the new page from being able to access the window.opener property and ensures it runs in a separate process.

> Without this the other page can potentially redirect your page to a malicious URL.

Reviewer: ```suggestion
Without this the target page can potentially redirect your page to a malicious URL.
```
?

`rel="noreferrer""` has the same effect, but also prevents the Referer header from being sent to the new page.

Removing the referer header will affect analytics. By stripping the referral value, the traffic from these links will be
misattributed — instead of showing as referral traffic, they will be attributed as “direct” in Google Analytics. Or, if

> you have another analytics program, you may see the visits in a noreferral bucket.

Reviewer: ```suggestion
you use an alternative analytics application, you may see the visits in a "noreferral" bucket.
```
